.github/workflows/haskell-ci.yml

```diff
@@ -50,22 +53,23 @@ jobs:
     strategy:
       matrix:
         stack:
-          - 2.13.1
+          - 3.1.1
         stack-yaml:
           - stack-9.0.2.yaml
           - stack-9.2.8.yaml
           - stack-9.4.8.yaml
-          - stack-9.6.3.yaml
+          - stack-9.6.6.yaml
+          - stack-9.8.3.yaml
```

Comment:
Stackage nightly has already 9.8.3 (but doesn't have 9.10.1), while ghcup only has 9.8.2 (but has 9.10.1).
I went for the latest available for both.